test/cqlpy/test_vector_similarity.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+import pytest
+from .util import new_test_table, is_scylla
+from cassandra.protocol import InvalidRequest
+import math
+
+
+###############################################################################
+# Tests for vector search related functions
+###############################################################################
+
+
+REQUIRES_FLOAT_VECTOR = "Function system.similarity_{} requires a float vector argument"
+
+
+def compute_similarity(similarity_function, v1, v2):
+    if similarity_function == "cosine":
+        dot = sum(a * b for a, b in zip(v1, v2))
+        norm_v = math.sqrt(sum(x**2 for x in v1))
+        norm_q = math.sqrt(sum(x**2 for x in v2))
+        if norm_v == 0 or norm_q == 0:
+            raise ValueError("Cosine similarity is not defined for zero vectors")
+        cosine = dot / (norm_v * norm_q)
+        return (1 + cosine) / 2
+    elif similarity_function == "euclidean":
+        euclidean_sq = sum((a - b)**2 for a, b in zip(v1, v2))
+        return 1 / (1 + euclidean_sq)
+    elif similarity_function == "dot_product":
+        dot_product = sum(a * b for a, b in zip(v1, v2))
+        return (1 + dot_product) / 2
+
+
+def assert_close(actual, expected):
+    assert math.isclose(actual, expected, abs_tol=1e-5)
```

Comment:
nitpick: there's no real need to have an assert_close function, you can have an isclose() function returning a boolean and then write
```
assert isclose(...)
```
This would be more pytest-like, and will also allow you to do things like `assert isclose(..) and ...` and so on.